wallet/how-to/interact-with-smart-contracts/unity/contract-interface.md

```diff
@@ -0,0 +1,150 @@
+---
+sidebar_position: 1
+---
+
+# Contract interface
+
+When [interacting with smart contracts in Unity](index.md), the `IContract` interface defines an
+interface of a given contract.
+This interface includes all the functions of the contract, and optionally includes a constructor
+function and `Bytecode` field.
+
+You can use the [contract code generator](index.md#generate-contract-code) to generate a contract
+interface given a contract ABI or Hardhat artifact JSON file.
+
+To manually create a new contract interface, you must define a new interface that inherits from `IContract`.
+You may optionally declare the [`BackedType` attribute](contract-factory.md#backed-type-contract-factory).
+
+```csharp
+#if UNITY_EDITOR || !ENABLE_MONO  
+[BackedType(typeof(ERC20Backing))]  
+#endif  
+public interface ERC20 : IContract  
+{
+	// TODO Declare functions
+}
+```
+
+## Declare contract functions
+
+To declare a `view` or `pure` function of the contract, first set the return type, the function name
+(may differ from the actual name), and all parameters the function takes in a `Task`:
+
+```csharp
+#if UNITY_EDITOR || !ENABLE_MONO  
+[BackedType(typeof(ERC20Backing))]  
+#endif  
+public interface ERC20 : IContract  
+{
+	Task<BigInteger> BalanceOf(EvmAddress account);
+}
+```
+
+Once you have the function written, simply add the `EvmMethodInfo` at the top of the function to
+declare the metadata about the contract function.
+This includes the `Name` and whether it's a `View` function:
+
+```csharp
+#if UNITY_EDITOR || !ENABLE_MONO  
+[BackedType(typeof(ERC20Backing))]  
+#endif  
+public interface ERC20 : IContract  
+{
+	[EvmMethodInfo(Name = "balanceOf", View = true)]
+	Task<BigInteger> BalanceOf(EvmAddress account);
+}
+```
+
+To define the EVM type for a parameter, you can use the `EvmParameterInfo` attribute.
+However, this usually isn't needed, because the `Contract` class automatically infers most common
+types, such as `EvmAddress` to be `address` and `string` to be `string`.
+
+```csharp
+#if UNITY_EDITOR || !ENABLE_MONO  
+[BackedType(typeof(ERC20Backing))]  
+#endif  
+public interface ERC20 : IContract  
+{
+	[EvmMethodInfo(Name = "balanceOf", View = true)]
+	Task<BigInteger> BalanceOf([EvmParameterInfo(Type = "address")] string account);
+}
+```
+
+To define the EVM return type for the function, you can use `EvmParamterInfo` on the return type:
+
+```csharp
+#if UNITY_EDITOR || !ENABLE_MONO  
+[BackedType(typeof(ERC20Backing))]  
+#endif  
+public interface ERC20 : IContract  
+{
+	[EvmMethodInfo(Name = "balanceOf", View = true)]
+	Task<BigInteger> BalanceOf(EvmAddress account);
+	
+	[EvmMethodInfo(Name = "decimals", View = true)]  
+	[return: EvmParameterInfo(Type = "uint8")]  
+	Task<BigInteger> Decimals();
+}
+```
+
+### Use the Task return type
+
+It's recommend to always use `Task` as the return type when declaring contract functions, even if
```

Comment:
Some say... "We recommend..." It's more active. 

If you keep it your way, then... recommend = recommended